src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx

```diff
@@ -86,11 +112,133 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
     expect((await screen.findAllByRole('radio')).length).to.equal(2);
     expect(screen.getByLabelText('Bozeman, MT')).to.exist;
     expect(screen.getByLabelText('Belgrade, MT')).to.exist;
+    expect(screen.baseElement).to.contain.text(
+      'Request a Primary care provider. (Optional)',
+    );
+    expect(screen.baseElement).to.contain.text(
+      'You can request a provider you’d prefer for this appointment. If they aren’t available, we’ll schedule your appointment with a provider close to your home.',
+    );
+    expect(screen.baseElement).to.contain.text('Choose a provider');
 
     // Continue without filling in required fields
     userEvent.click(screen.getByText(/Continue/i));
 
     expect((await screen.findAllByRole('alert')).length).to.equal(2);
     expect(screen.history.push.called).to.be.false;
   });
+
+  it('should display list of providers when choose a provider clicked', async () => {
+    const store = createTestStore(initialState);
+    await setTypeOfCare(store, /primary care/i);
+    await setTypeOfFacility(store, /Community Care/i);
+    const screen = renderWithStoreAndRouter(
+      <CommunityCareProviderSelectionPage />,
+      {
+        store,
+      },
+    );
+
+    // Trigger provider list loading
+    userEvent.click(await screen.findByText(/Choose a provider/i));
+    await waitFor(() => screen.getByText(/your address on file:/i));
+
+    // Verify provider list count and get load more button
+    expect(screen.baseElement).to.contain.text(
+      '123 big sky stCincinnati, OH 45220',
+    );
+    expect(screen.baseElement).to.contain.text(
+      'Displaying 1 to 5 of 16 providers',
+    );
+    expect((await screen.findAllByRole('radio')).length).to.equal(7);
+
+    userEvent.click(await screen.findByText(/\+ 5 more providers/i));
+    await waitFor(() =>
+      screen.getByText(/displaying 1 to 10 of 16 providers/i),
+    );
+    expect((await screen.findAllByRole('radio')).length).to.equal(12);
+
+    userEvent.click(await screen.findByText(/\+ 5 more providers/i));
+    await waitFor(() =>
+      screen.getByText(/displaying 1 to 15 of 16 providers/i),
+    );
+    expect((await screen.findAllByRole('radio')).length).to.equal(17);
+
+    userEvent.click(await screen.findByText(/\+ 1 more providers/i));
+    await waitFor(() =>
+      screen.getByText(/displaying 1 to 16 of 16 providers/i),
+    );
+    expect((await screen.findAllByRole('radio')).length).to.equal(18);
+
+    // Choose Provider
+    userEvent.click((await screen.findAllByRole('radio'))[3]);
```

Comment:
Sort of nitpicky, but we'd probably prefer to pick the radio button by doing `findByLabelText`, and providing a regex with part of the provider name (or whatever is in the label), since that's more analogous to what a user would do.

Reply:
Fixed